src/main/scala/com/fortysevendeg/scalacheck/datetime/j8/GenJ8.scala

```diff
@@ -0,0 +1,26 @@
+package com.fortysevendeg.scalacheck.datetime.j8
+
+import collection.JavaConverters._
+
+import org.scalacheck.Gen
+import org.scalacheck.Arbitrary.arbitrary
+
+import java.time._
+import java.time.temporal.ChronoUnit.MILLIS
+
+object GenJ8 {
+
+  val genZonedDateTime: Gen[ZonedDateTime] = for {
+    year <- Gen.choose(-292278994, 292278994)
+    month <- Gen.choose(1, 12)
+    maxDaysInMonth = Month.of(month).length(Year.of(year).isLeap)
+    dayOfMonth <- Gen.choose(1, maxDaysInMonth)
+    hour <- Gen.choose(0, 23)
```

Comment:
Great comment, thanks. This is something I did think about, and corner-cases like this are really the motivation for the library.

I've just tried to create a time with `24:00` - and it blew up with an exception.
Also, I just tried to create a time for a valid leap-second - and again it blew up with an exception.

The `java.time` documentation is explicit in its inputs: https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html#of-int-int-int-int-int-int-int-java.time.ZoneId-

It looks like `java.time` doesn't handle this right now.